sim/rogue/assassination/assassination.go

```diff
@@ -23,11 +27,60 @@ func RegisterAssassinationRogue() {
 	)
 }
 
+func (sinRogue *AssassinationRogue) Initialize() {
+	sinRogue.Rogue.Initialize()
+
+	sinRogue.registerMutilateSpell()
+	sinRogue.registerOverkill()
+	sinRogue.registerColdBloodCD()
+	sinRogue.applySealFate()
+	sinRogue.registerVenomousWounds()
+	sinRogue.registerVendetta()
+
+	// Apply Mastery
+	masteryPercent := sinRogue.GetStat(stats.Mastery) / core.MasteryRatingPerMasteryPoint
+	masteryEffect := masteryBaseEffect + masteryPercent*masteryDamagePerPercent
+	for _, spell := range sinRogue.InstantPoison {
+		spell.DamageMultiplier += masteryEffect
+	}
+	for _, spell := range sinRogue.WoundPoison {
+		spell.DamageMultiplier += masteryEffect
+	}
+	sinRogue.DeadlyPoison.DamageMultiplier += masteryEffect
+	sinRogue.Envenom.DamageMultiplier += masteryEffect
+	if sinRogue.Talents.VenomousWounds > 0 {
+		sinRogue.VenomousWounds.DamageMultiplier += masteryEffect
+	}
+
+	sinRogue.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery, newMastery float64) {
+		masteryPercentOld := oldMastery / core.MasteryRatingPerMasteryPoint
+		masteryPercentNew := newMastery / core.MasteryRatingPerMasteryPoint
+		masteryEffectChange := masteryBaseEffect + (masteryPercentNew-masteryPercentOld)*masteryDamagePerPercent
+		for _, spell := range sinRogue.InstantPoison {
+			spell.DamageMultiplier += masteryEffectChange
+		}
+		for _, spell := range sinRogue.WoundPoison {
+			spell.DamageMultiplier += masteryEffectChange
+		}
+		sinRogue.DeadlyPoison.DamageMultiplier += masteryEffectChange
+		sinRogue.Envenom.DamageMultiplier += masteryEffectChange
+		if sinRogue.Talents.VenomousWounds > 0 {
+			sinRogue.VenomousWounds.DamageMultiplier += masteryEffectChange
+		}
```

Comment:
This would be wrong the moment another multiplier is applied inbetween mastery changes.

Instead you should calculate old mastery bonus and remove it from the multiplier then calculate new mastery bonus and add it to the multiplier.